spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/DistributionAndOrderingUtils.scala

```diff
@@ -118,7 +120,24 @@ object DistributionAndOrderingUtils {
     expr match {
       case s: SortOrder =>
         val catalystChild = toCatalyst(s.expression(), query, resolver)
-        catalyst.expressions.SortOrder(catalystChild, toCatalyst(s.direction), toCatalyst(s.nullOrdering), Set.empty)
+        val ctor: DynConstructors.Ctor[catalyst.expressions.SortOrder] =
+          DynConstructors.builder()
+            .impl(classOf[catalyst.expressions.SortOrder],
+              classOf[catalyst.expressions.Expression],
+              classOf[catalyst.expressions.SortDirection],
+              classOf[catalyst.expressions.NullOrdering],
+              classOf[Seq[catalyst.expressions.Expression]])
+            .impl(classOf[catalyst.expressions.SortOrder],
+              classOf[catalyst.expressions.Expression],
+              classOf[catalyst.expressions.SortDirection],
+              classOf[catalyst.expressions.NullOrdering],
+              classOf[Set[catalyst.expressions.Expression]])
+            .build()
+        if (Spark3VersionUtil.isSpark30) {
+          ctor.newInstance(catalystChild, toCatalyst(s.direction), toCatalyst(s.nullOrdering), Set.empty)
```

Comment:
Does `Set` implement `Seq`? If so, does it work to just use the `Set.empty` variation here?

Reply:
`Set` does not extend `Seq` nor does `Seq` extend `Set`. I don't know of any common subclass of both in the standard Scala library.